packages/react-app/src/views/BrowseBadges.jsx

```diff
@@ -147,19 +155,52 @@ export default function BrowseBadges() {
 
   const run = useCallback(async () => {
     if (address) {
-      return setEventBadges([])
+      setEventBadges([])
+      return
     }
     let badges = await getAllRewards(contractRef.address, providerRef)
-    // console.log({ badges })
     badges = badges.map(badge => {
       return {
         id: ethers.utils.hexStripZeros(badge.topics[3]),
         to: ethers.utils.hexZeroPad(ethers.utils.hexStripZeros(badge.topics[2]), 20),
         transactionHash: badge.transactionHash,
       }
     })
+    let dataArray = []
+    const result = badges.map(async badge => {
+      if (badge.id === '0x') {
+        badge.id = '0x0'
+      }
+      let temp = { ...badge }
+
+      let data = await contract.current.tokensData(badge.id)
+      temp.resolvedName = await mainnet.lookupAddress(temp.to)
+      dataArray.push(data)
+      for (let index = 0; index < data.length; index++) {
+        if (data[index].length > 20) {
```

Comment:
why do you do these check? if it's necessary can you add a comment in the code?

Reply:
ok